Bio/codonalign/codonseq.py

```diff
@@ -265,30 +239,19 @@ def full_translate(self, codon_table=default_codon_table, stop_symbol="*"):
 
     def ungap(self, gap=None):
         """Return a copy of the sequence without the gap character(s)."""
-        if hasattr(self.alphabet, "gap_char"):
-            if not gap:
-                gap = self.alphabet.gap_char
-            elif gap != self.alphabet.gap_char:
-                raise ValueError(
-                    "Gap %s does not match %s from alphabet"
-                    % (repr(gap), repr(self.alphabet.alphabet.gap_char))
-                )
-            alpha = _ungap(self.alphabet)
-        elif not gap:
-            raise ValueError("Gap character not given and not defined in alphabet")
-        else:
-            alpha = self.alphabet  # modify!
+        if gap is None:
+            raise ValueError("Gap character not given")
```

Comment:
If the default gap is '-', then we don't need to do this check here.

Reply:
Fixed, thank you.